sdk/storage/AzureStorageBlob/Source/DataStructures/URLSessionTransferManager.swift

```diff
@@ -0,0 +1,389 @@
+            for transfer in transfers {
+                context.delete(transfer)
+            }
+        }
+        let blobRequest: NSFetchRequest<BlobTransfer> = BlobTransfer.fetchRequest()
+        if let transfers = try? context.fetch(blobRequest) {
+            for transfer in transfers {
+                context.delete(transfer)
+            }
+        }
+        let blockRequest: NSFetchRequest<BlockTransfer> = BlockTransfer.fetchRequest()
+        if let transfers = try? context.fetch(blockRequest) {
+            for transfer in transfers {
+                context.delete(transfer)
+            }
+        }
+        operations(nil, didChangeState: .deleted)
+    }
+
+    public func remove(transfer: Transfer) {
+        switch transfer {
+        case let transfer as BlockTransfer:
+            remove(transfer: transfer)
+        case let transfer as BlobTransfer:
+            remove(transfer: transfer)
+        default:
+            fatalError("Unrecognized transfer type: \(transfer.self)")
+        }
+        transfer.state = .deleted
+        operation(transfer.operation, didChangeState: .deleted)
+    }
+
+    internal func remove(transfer: BlockTransfer) {
+        if let operation = transfer.operation {
+            operationQueue.remove(operation)
+        }
+
+        if let index = transfers.firstIndex(where: { $0 === transfer }) {
+            transfers.remove(at: index)
+        }
+
+        // remove the object from CoreData
+        if let context = persistentContainer?.viewContext {
+            context.delete(transfer)
+        }
+    }
+
+    internal func remove(transfer: BlobTransfer) {
+        // Cancel the operation and any associated block operations
+        if let operation = transfer.operation {
+            operationQueue.remove(operation)
+            for block in transfer.transfers {
+                block.state = .deleted
+                if let blockOp = block.operation {
+                    operationQueue.remove(blockOp)
+                }
+            }
+        }
+
+        // Remove the blob operation from the transfers list
+        if let index = transfers.firstIndex(where: { $0 === transfer }) {
+            transfers.remove(at: index)
+        }
+
+        // remove the object from CoreData which should cascade and delete any outstanding block transfers
+        if let context = persistentContainer?.viewContext {
+            context.delete(transfer)
+        }
+    }
+
+    // MARK: Pause Operations
+
+    public func pause(transfer: Transfer) {
+        let allowed: [TransferState] = [.pending, .inProgress]
+        guard allowed.contains(transfer.state) else { return }
+
+        switch transfer {
+        case let transfer as BlockTransfer:
+            pause(transfer: transfer)
+        case let transfer as BlobTransfer:
+            pause(transfer: transfer)
+        default:
+            fatalError("Unrecognized transfer type: \(transfer.self)")
```

Comment:
Could this be a MultiBlobTransfer? And if so would it hit this fatalError case?

Reply:
Yes, there are two issues here:
1) MultiBlobTransfer is not fully implemented (and I don't plan to for the UX study)
2) it shouldn't fatalError regardless.